src/system-update-generator/system-update-generator.c

```diff
@@ -18,19 +19,22 @@
 static const char *arg_dest = NULL;
 
 static int generate_symlink(void) {
-        const char *p = NULL;
+        _cleanup_free_ char *j = NULL;
```

Comment:
You can move this into `link_found`.

(Or you can avoid `goto` by using a `bool found`)

Reply:
I left the goto in and I also left the declaration at the top, gcc warns when you move to link_found